sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,364 @@ def to_rotation_matrix(self, v=None):
 
             return Matrix([[m00, m01, m02, m03], [m10, m11, m12, m13],
                           [m20, m21, m22, m23], [m30, m31, m32, m33]])
+
+    def scalar(self):
+        """
+        Returns the scalar part of the quaternion.
+
+        Explanation
+        ===========
+
+        The scalar component of a quaternion is the real part of the quaternion.
+        It gives the rotation angle, and the three imaginary components, when
+        normalized, give the direction ofthe rotation axis.
```

Comment:
I don't think this really explains what the scalar part is at all. A short unambiguous, explicit definition is what is needed like:

If q is a quaternion given by `q = a + b*i + c*j + d*k` where a, b, c and d are real numbers then the scalar part of `q` is `a`.

Then there can be a see also for the vector part.

Reply:
Maybe this should be called `scalar_part`. Maybe also/instead it makes sense to have a method that returns both the scalar part and the vector part like `s, v = q.as_scalar_vector()`

Reply:
The word "part" would be redundant here. There's nothing wrong in including another façade `s, v = q.as_scalar_vector()`, I liked it, but it doesn't replace the segregated methods to get either the scalar or the vector part of a quaternion. `S(q)` and `V(q)` are quite common separately when working with quaternions (and indeed required in order to properly work with quaternions). These are like vector projections, sometimes we just need the projection, not the "orthogonal residual" (or vice-versa), getting both sometimes makes sense, but that's not the general case.

Reply:
Is it common to denote these as `S(q)` and `V(q)`? I haven't seen that notation before.

Reply:
It's probably used by every single author since 1847, or most of them at least, until Gibbs/Heaviside proposed the dot/cross products and some other stuff just to avoid quaternions, moving part of its jargon and results to their "vector algebra". The cross product of two vectors `q` and `r` is the vector part of the quaternion product `V(qr)` when `q` and `r` are pure quaternions (3D vectors), that's really the origin of the cross product, and a quite common use case where `V(·)` is required but `S(·)` isn't. Hamilton himself used `Sq` and `Vq`, and `1 = S + V` since 1846 to mean `S(q)`, `V(q)` and `q = S(q) + V(q)` in our current notation, as at that time (1840's) the parentheses weren't in use (they used to put a dot to denote the end of the parameter, so `SrSq` is `S[rS(q)]` while `Sr.Sq` is `S(r)S(q)` ), unless required, e.g. because of a sum like S(q+r).

Reply:
After Gibbs/Heaviside argument regarding the Maxwell's equations, quaternions were almost forgotten. Their classical notation didn't change, what happened is that quaternions almost disappeared. Because of SLERP for animation/games graphics, some people are saying they're using quaternions nowadays, but most people using them aren't really trying to understand them, quaternions became just a tool (that's not really a problem, unless they try to argue something regarding the theory without knowing it). Actually, SLERP is quite simple for anyone who knows that quaternions are general "arcs", it's the same to `(b/a)**t a` for complex numbers `a` and `b`, and `t` from 0 to 1: a linear interpolation of the angle. Quaternions do the same in 3D. Assuming this interpolation can be written a linear combination of the input one can find the coefficients in a quite straightforward way, without relying on Gibbs' algebra. But any use of a cross product together with quaternions to explain quaternions is historically misleading and might lose interpretation: cross products came from quaternions, not the other way around. On the other hand, thinking on Clifford's algebra would help on understand quaternions, since it came from embracing the quaternion multiplication, the "scalar + vector" idea, and Grassman's exterior/outer product, not from avoiding these.

As a summary, "cross product" should not appear together with quaternions, and the need of a cross product is a possible justification for a `V(q)` appearing without `S(q)`, and that's just one simple use case, far from being the only one.